pkg/discovery/utils.go

```diff
@@ -49,31 +50,14 @@ func FilterNamespaces(kubeClient kubernetes.Interface, filter string) ([]string,
 // SerializeObj will write out an object
 func SerializeObj(obj interface{}, outpath string, file string) error {
 	var err error
-	if err = os.MkdirAll(outpath, 0755); err == nil {
-		if eJSONBytes, err := json.Marshal(obj); err == nil {
-			err = ioutil.WriteFile(outpath+"/"+file, eJSONBytes, 0644)
-		}
+	if err = os.MkdirAll(outpath, 0755); err != nil {
```

Comment:
These changes to the serializers were the result of tearing out a bunch and putting back in what I needed. TL; DR; we just didnt need all these functions anymore and the ones we did could be written a bit more idiomatically to make it easier to read.